app/src/common/shared/com/igalia/wolvic/browser/engine/SessionStore.java

```diff
@@ -514,28 +529,43 @@ public void removePermissionException(@NonNull String uri, @SitePermission.Categ
     @Override
     public void onSessionAdded(Session aSession) {
         ComponentsAdapter.get().addSession(aSession);
+        for (SessionChangeListener listener : mSessionChangeListeners) {
```

Comment:
A shared question for all those methods. The Session object do implement the SessionChangeListener interface already. Why not using it, instead of adding it to the SessionStore? Seems unnecessarily complex and make the whole app flow more difficult to follow.

Reply:
Hmm after thinking about it twice, I guess it's because the tabs are "shared" and thus we cannot just get the session of a specific window. Is that the rationale?

Reply:
Yes, that's basically the reason. There is only one SessionStore with one list of open tabs, and one tab widget that displays them.

I had to do it this way so the widget could react to changes in the list of tabs.